File: book/2e/data/ch04/fizzbuzz.R
#!/usr/bin/env Rscript
cycle_of_15 <- c("fizzbuzz", NA, NA, "fizz", NA,
                 "buzz", "fizz", NA, NA, "fizz",
                 "buzz", NA, "fizz", NA, NA)

fizz_buzz <- function(n) {
  word <- cycle_of_15[as.integer(n) %% 15 + 1]
  ifelse(is.na(word), n, word)
}

f <- file("stdin")
open(f)
while(length(n <- readLines(f, n = 1)) > 0) {
  write(fizz_buzz(n), stdout())
}
close(f)

File: book/2e/data/ch04/fizzbuzz.py
#!/usr/bin/env python
import sys

CYCLE_OF_15 = ["fizzbuzz", None, None, "fizz", None,
               "buzz", "fizz", None, None, "fizz",
               "buzz", None, "fizz", None, None]

def fizz_buzz(n: int) -> str:
    return CYCLE_OF_15[n % 15] or str(n)

if __name__ == "__main__":
    try:
        while (n:= sys.stdin.readline()):
            print(fizz_buzz(int(n)))
    except:
        pass
